Added IPFS #3

RichardLitt opened the pull request:
This is a bit late because I was hoping to have `name-your-contributors` done, but I didn't manage to do it, and now I need to add a timespan. 

@eminence, can you add the contribs for last week? 

Review period open for this Friday.

RichardLitt committed:
Added 003

dignifiedquire commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,27 @@
+
+## Updates
+
+* [go-ipfs](//github.com/ipfs/go-ipfs) 0.3.11 is shipped! Also, we merged 0.4.0 into master. This is huge news that we've been waiting for for a long time. Also, [@jbenet](//github.com/jbenet), [@Dignifiedquire](//github.com/Dignifiedquire) and others [fixed appveyor](//github.com/ipfs/go-ipfs/pull/2137). The latest webui by [@Dignifiedquire](//github.com/Dignifiedquire) was also pushed in 0.3.11 ...
```
repetition of `also` in the beginning of a sentance

dignifiedquire replied:
`This is huge news that we've been waiting for for a long time.` This sentence sounds strange :/ (sorry for not being more helpful)

dignifiedquire commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,27 @@
+
+Across the entire IPFS GitHub organization, the following people have committed code since January 4th. (We're autogenerating this list using [this tool](//github.com/ipfs/weekly/blob/master/tools/get_commits.py), so please let us know if your name isn't here.) In the future, we will also include people who comment, as they are also super important; bear with us while we develop that technology.
+
+// TODO Add this
```
This is not rendering properly

RichardLitt replied:
Needs to be fixed before shipping. We need to add contribs. Working on it.

dignifiedquire commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,27 @@
+* [ipfs-merkle-dag](//github.com/vijayee/ipfs-merkle-dag) [@daviddias](//github.com/daviddias) worked on the js ipfs merkle-dag implementation, as well as putting in a lot of work in general on js-ipfs.
+
+## Work in Progress
+* [weekly](//github.com/ipfs/weekly) [@RichardLitt](//github.com/RichardLitt) put a lot of work in getting various sorts of attribution for contributors to IPFS, currently available as [name-your-contributors](//github.com/RichardLitt/name-your-contributors). This is almost done, but there's a few tweaks left to do.
```
`there are a few tweaks`

jbenet commented:
There's likely more stuff that got done last week that needs to be mentioned here. 

Look harder in the sprint issues, don't just limit yourself to starred things. And ask people directly if you dont have enough stuff listed, or are not sure whether to highlight something. People are starting to depend on this weekly to measure the heartbeat of the community and it should show accurate progress. Check also IRC logs, sometimes there's stuff posted worth highlighting. (and can also direct those people to post such cool stuff in the sprint issues in the future)

eminence committed:
Add commit contribs

(Mon Jan 11 00:00:00 2016 to Tue Jan 19 00:00:00 2016)

I added 1 day, due to the 1 day delay on monday

RichardLitt commented:
Thanks @jbenet. It's been difficult to find the time for this over the past few days, with traveling. @whyrusleeping can you show me how I can access the IRC logs? I think the weekly would be better to be a as-we-go thing instead of an end of week summary, will make the process easier. 

As it is, I'm not sure how much else there is; I asked @whyrusleeping, @noffle, and @diasdavid directly with little feedback about stuff to add. :/

whyrusleeping commented:
I can send you my irc logs, but its just going to be a static copy of my weechat logs.

I can't really think of anything that happened this last week thats worthy of being put on the weekly... I'll post my sprint update in a little bit if you want to grep that for stuff

whyrusleeping commented:
oh, right this is the previous week. got it.

Dignifiedquire commented:
Here are the logs for ipfs IRC https://botbot.me/freenode/ipfs/

eminence commented:
I too have been busy with other things, and haven't contributed much to this weeks writeup.   I can support @RichardLitt's idea of turning this into a "whenever needed" affair.   When I first proposed these weekly writeups, I had hoped that they would be fairly quick to produce.  The ones we've produced so far are really great!  But I admit that I'm slightly concerned that the multi-hour effort to produce one of these is not sustainable.

RichardLitt committed:
Added all of the contribs

RichardLitt committed:
Change dates

RichardLitt committed:
Added content from second week

RichardLitt committed:
Added note about double weekly

daviddias commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,120 @@
+* [**py-ipfs**](//github.com/ipfs/py-ipfs) Thanks to [@candeira](//github.com/candeira) and [@ivilata](//github.com/ivilata), we now have a much more filled out plan for the Python implementation of IPFS. Join in if you are a pythonista.
+* [**js-mafmt**](//github.com/whyrusleeping/js-mafmt) [@whyrusleeping](//github.com/whyrusleeping) is working on a multiaddr validator in js, after his go work in [mafmt](//github.com/whyrusleeping/mafmt)
+* [**archives**](//github.com/ipfs/archive) cdn.media.ccc.de is now being mirrored in the IPFS archives.
+* [**ifps-firefox-addon**](//github.com/lidel/ipfs-firefox-addon/releases/tag/v1.4.2) The Firefox addon that provides transparent access to IPFS resources via local HTTP2IPFS gateway has been updated to release 1.4.2. This is the first time we've mentioned this in this newsletter; go check it out. It also now supports the fs: protocol scheme.
```
ipfs*

RichardLitt committed:
Update spelling error

RichardLitt committed:
Added the changelog

RichardLitt committed:
Fix arcticjs link

RichardLitt committed:
Merge github.com:ipfs/weekly into feature/jan-11

RichardLitt committed:
Added note about infrastructure from @lgierth

RichardLitt committed:
Remove premature file

RichardLitt committed:
Added go-multiaddr and links to ArcticJS talks

RichardLitt committed:
Added a lot of things from https://github.com/ipfs/pm/issues/83

RichardLitt committed:
Remove dupe, diasdavid has an annoying name

RichardLitt committed:
Update feedback link to PR

jbenet commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,127 @@
+## Updates
+
+* [**ArcticJS**](http://arcticjs.club) [@diasdavid](//github.com/diasdavid), [@whyrusleeping](//github.com/whyrusleeping), [@RichardLitt](//github.com/RichardLitt), and [@noffle](//github.com/noffle) were all able to meet up for the first time as a group and hack together at [ArcticJS](http://arcticjs.club), the first JavaScript conference above the Arctic circle, in Svalbard. Around 15 attended, ...
+* [**dist.ipfs.io**](https://dist.ipfs.io) A distributions page has been shipped! This was largely spearheaded by [@dignifiedquire](//github.com/dignifiedquire). http://dist.ipfs.io/
```
expand on this, this is way more important for us than ArticJS. maybe @dignifiedquire can help.

jbenet commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,127 @@
+
+## Updates
+
+* [**ArcticJS**](http://arcticjs.club) [@diasdavid](//github.com/diasdavid), [@whyrusleeping](//github.com/whyrusleeping), [@RichardLitt](//github.com/RichardLitt), and [@noffle](//github.com/noffle) were all able to meet up for the first time as a group and hack together at [ArcticJS](http://arcticjs.club), the first JavaScript conference above the Arctic circle, in Svalbard. Around 15 attended, ...
```
this is awesome, maybe put it under it's own heading, like "Conferences" or something.

jbenet replied:
Or "Meetups + Conferences" or something

RichardLitt replied:
Moved to it's own section.

jbenet commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,127 @@
+
+* [**ArcticJS**](http://arcticjs.club) [@diasdavid](//github.com/diasdavid), [@whyrusleeping](//github.com/whyrusleeping), [@RichardLitt](//github.com/RichardLitt), and [@noffle](//github.com/noffle) were all able to meet up for the first time as a group and hack together at [ArcticJS](http://arcticjs.club), the first JavaScript conference above the Arctic circle, in Svalbard. Around 15 attended, ...
+* [**dist.ipfs.io**](https://dist.ipfs.io) A distributions page has been shipped! This was largely spearheaded by [@dignifiedquire](//github.com/dignifiedquire). http://dist.ipfs.io/
+* [**go-ipfs**](//github.com/ipfs/go-ipfs) 0.3.11 is shipped! Also, we merged 0.4.0 into master. [@jbenet](//github.com/jbenet), [@Dignifiedquire](//github.com/Dignifiedquire) and others [fixed appveyor](//github.com/ipfs/go-ipfs/pull/2137), which means that windows CI now works. And finally the latest webui by [@Dignifiedquire](//github.com/Dignifiedquire) was pushed in 0.3.11. Check out the [ch ...
```
0.4.0 means much more than "windows CI now works". i'd just remove that part, or attempt at describing 0.4.0's main improvements. (the windows CI things is a tiny factor comparatively)

RichardLitt replied:
Removed note.

jbenet commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,127 @@
+* [**ipfs-merkle-dag**](//github.com/vijayee/ipfs-merkle-dag) [@diasdavid](//github.com/diasdavid) worked on the js ipfs merkle-dag implementation, as well as putting in a lot of work in general on js-ipfs.
+* [**py-ipfs**](//github.com/ipfs/py-ipfs) Thanks to [@candeira](//github.com/candeira) and [@ivilata](//github.com/ivilata), we now have a much more filled out plan for the Python implementation of IPFS. Join in if you are a pythonista.
+* [**js-mafmt**](//github.com/whyrusleeping/js-mafmt) [@whyrusleeping](//github.com/whyrusleeping) is working on a multiaddr validator in js, after his go work in [mafmt](//github.com/whyrusleeping/mafmt)
+* [**infrastructure**](//github.com/ipfs/infrastructure) The public gateway at [ipfs.io](https://ipfs.io) is now backed by both v0.4.0-dev and v0.3.11 at the same time. Requests are proxied to both, and the first successful response wins. ipfs.io will be backed by v0.4.x alone eventually. What this means is that if you want to target your requests at one specific version, use v04x.ipfs.io or v03x ...
```
@lgierth it may be important to write more in an md doc what the the current state / future plans are (maybe as a file in infrastructure repo), and link to it here.

jbenet commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,127 @@
+* [**py-ipfs**](//github.com/ipfs/py-ipfs) Thanks to [@candeira](//github.com/candeira) and [@ivilata](//github.com/ivilata), we now have a much more filled out plan for the Python implementation of IPFS. Join in if you are a pythonista.
+* [**js-mafmt**](//github.com/whyrusleeping/js-mafmt) [@whyrusleeping](//github.com/whyrusleeping) is working on a multiaddr validator in js, after his go work in [mafmt](//github.com/whyrusleeping/mafmt)
+* [**infrastructure**](//github.com/ipfs/infrastructure) The public gateway at [ipfs.io](https://ipfs.io) is now backed by both v0.4.0-dev and v0.3.11 at the same time. Requests are proxied to both, and the first successful response wins. ipfs.io will be backed by v0.4.x alone eventually. What this means is that if you want to target your requests at one specific version, use v04x.ipfs.io or v03x ...
+* [**ifps-firefox-addon**](//github.com/lidel/ipfs-firefox-addon/releases/tag/v1.4.2) The Firefox addon that provides transparent access to IPFS resources via local HTTP2IPFS gateway has been updated to release 1.4.2. This is the first time we've mentioned this in this newsletter; go check it out. It also now supports the IPFS protocol scheme.
```
include a sentence or two about what this addon does.

jbenet replied:
cc @lidel

lidel replied:
Release page for 1.4.2 does not tell much :) Perhaps it is a better idea for the name "ipfs-firefox-addon" to be a  link to https://github.com/lidel/ipfs-firefox-addon#ipfs-firefox-addon? 

Then the phrase "release 1.4.2" could link to https://github.com/lidel/ipfs-firefox-addon/releases/tag/v1.4.2

RichardLitt replied:
Done!

jbenet commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,127 @@
+* [**go-multiaddr**](//github.com/jbenet/go-multiaddr) go-multiaddr now supports the adding of new protocols on the fly. This allows us to build in new protocols without having to force everyone to update the multiaddr library itself.
+* [**community**](//github.com/ipfs/community) The labels 'help wanted' and 'difficulty:easy', ':hard', and ':moderate' have been added to every IPFS repository, to help out contributors know what are some low-hanging fruit they can help with. As well, [@RichardLitt](//github.com/RichardLitt) has created a [Docs Styleguide](https://github.com/ipfs/community/blob/master/docs-styleguide.md) to help ...
+* [**ipfs-dht**](//github.com/whyrusleeping/ipfs-dht) [@whyrusleeping](//github.com/whyrusleeping) wrote a 'just a dht' binary. This means you can help out the network without running a full ipfs node. It also takes less memory and bandwidth, and having more dht nodes will speed up lookups.
+* [**js-ipfs-merkle-dag**](//github.com/vijayee/js-ipfs-merkle-dag/) [@diasdavid](//github.com/diasdavid) worked on merkle-dag implementation interoperabolity, by enabling you to successfully read (and write) MerkleDAG nodes that were stored into an IPFS Repo through go-ipfs, [in JavaScript](https://github.com/vijayee/js-ipfs-merkle-dag/pull/6).
```
i think this is here twice

RichardLitt replied:
Resolved.

jbenet commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,127 @@
+- [@windemut](//github.com/windemut)
+- [@wking](//github.com/wking) (W. Trevor King)
+- [@xicombd](//github.com/xicombd) (Francisco Baio Dias)
+- [@zignig](//github.com/zignig) (Simon Kirkby)
```
awesome list!

jbenet commented on published/003-2016-Jan-12.md:
```diff
@@ -0,0 +1,127 @@
+- [@fazo96](//github.com/fazo96) (Enrico Fasoli)
+- [@findkiko](//github.com/findkiko)
+- [@Foxcool](//github.com/Foxcool) (Alexander Babenko)
+- [@GitCop](//github.com/GitCop)
```
lol

jbenet replied:
(i kinda like it, credit where credit is due)

RichardLitt replied:
![bots are people too](http://www.opticalsloth.com/wp-content/uploads/2010/07/robotsarepeople11.jpg)

lidel replied:
http://pbfcomics.com/274/

RichardLitt committed:
Moved arcticjs

RichardLitt committed:
Removed comment in go-ipfs, dupe js-ipfs-merkle-dag

jbenet commented:
Possible to add:

- Joe Armstrong, inventor of Erlang, [mentioned](https://twitter.com/joeerl/status/692065172609994752) the importance of permanence and IPFS in a recent, and fantastic [series of posts about organizing software](http://joearms.github.io/2016/01/26/The-Unintentional-Side-Effects-of-a-Bad-Concurrency-Model.html). Thanks for the kind mention, Joe!

RichardLitt committed:
Added more about dist

RichardLitt committed:
Added note about Joe Armstrong

RichardLitt committed:
Added more desc to ipfs-firefox-addon with desc from @lidel

RichardLitt commented:
@jbenet Added note about Joe Armstrong, and also dealt with all comments except for the one about infrastructure blogpost. Can we publish this, you think?

lgierth commented:
I'll take care of v03x/v04x in a blog post

jbenet commented:
LGTM thanks guys
